### ISSUE TITLE
Поломанный форум

### ISSUE THREAD

**Issue by DjonyBastone** [bem-site/bem-forum-content-ru#1213]
Приветствую всех на форуме! 

Это мой первый пост, в течении нескольких дней хотел обратиться с возникающими вопросами, но не получалось.
В двух браузерах не работает пост и сортировка форума.

https://youtu.be/rO1k3j6PN8w

На видео записал экран + панель разработчика. OS Win10
Проблема на клиенте или сервере? У кого ни будь такое происходит?

EDGE, Chrome - отказываются добавлять пост. Chrome - постит только после того, как из панели разработчика запустил эмуляцию мобильника. Данный пост смог написать только с IE11.
Тревожно на свой проект тянуть - т.к. с JS мало знаком, и самостоятельно исправить не смогу. 

Ребята, дайте обратную связь...

**Comment by vithar** [comment]
Да, у нас взбесился JS на старой части сайта, к которой относится и форум тоже. Пока совсем не понятно в чём проблема.

**Comment by vithar** [comment]
Этой проблемы нет с новыми версиями bem-core 4.x + bem-components 5.x, так что на своём проекте можно использовать их свободно.

**Comment by tadatuta** [comment]
Эта проблема в принципе не в коде компонентов, а на стороне форума.
Код дичайше древний, его нужно целиком менять.

**Comment by DjonyBastone** [comment]
Спасибо. А ответ на пост с Хрома пишу))

Ребята, В верстке вообще новичок. Пока пробовал освоить БЭМ, научился верстать :D 

Как человек мало понимающий всего БЭМ (за пределами html, css) опишу с чем столкнулся:

- Информация по сборщикам разрозненная по гитхаб, было бы удобно по-пунктам в одном месте расписать - что от чего зависит, что и как надо настраивать. Понятно что ENB гибкий инструмент, но хотя-бы тезисно расписать что настраивается, и что в Project-stub настроено. Есть идея древовидно подать "сборщики", с их зависимостями, и деревом настройки плагинов и библиотек. 

- Уровни переопределения (desktop, touch), вместе с идеей деления по уровням, намекните хотя бы реализацию определения устройств (с какого устройства зашли). Именно как вы определяете, и что делаете если JS выключен у клиента...

- Как добавить свою тему оформления.

Понятно если перечитать по несколько раз документацию и форум, можно разобраться. Но эти вопросы повторяются не только у меня. Было бы не плохо добавить на сайт эту информацию.

Спасибо за методологию, надеюсь на развитие юзабилити в данной теме

**Comment by tadatuta** [comment]
> расписать что настраивается, и что в Project-stub настроено

Потенциально настраивается все, что угодно. Что-то просто, что-то сложнее, для чего-то потребуется написать свои технологии, но так или иначе практически любое требование можно удовлетворить.

В `project-stub` можно запустить `enb server`, открыть `localhost:8080` — нарисуется граф сборки, который более-менее должен дать представление о том, что там происходит под капотом. Еще можно посмотреть https://www.youtube.com/watch?v=-un-YYgU6Pg

> определения устройств (с какого устройства зашли)

Сервер парсит строку user agent-а и по ней определяет браузер и его фичи, на основе которых принимает решение, какой именно вариант верстки отдать.

> что делаете если JS выключен у клиента

Чаще всего — ничего. Если пользователь выключил JS, у него большая часть интернета не работает и он сам понимает, зачем ему такая радость. Статистически таких пользователей пренебрежительно мало. Впрочем, на самых важных сервисах у нас есть специальные версии для древних браузеров, где JS хотя и не отключен, но практически ничего не умеет, поэтому используется по минимуму и все основные задачи решаются без JS.

> Как добавить свою тему оформления

Самый простой способ — это скопировать существующую под новым именем, написать своих стилей, а потом указать новую тему в BEMJSON и deps.js.

**Comment by DjonyBastone** [comment]
Спасибо.

Но, исследуя граф который в окне bash - не ясно, что и за что отвечает. Вообще сложно воспринимать граф из символов. В ваших презентациях - красивый граф был.

К примеру реализация JS не правильно работает в project-stub для:
 - Собираем статическую страницу   //Имя пользователя не заменяется, скорее всего не правильно JS собирается.
 - Создаём свой проект на БЭМ   //Блок Box - не сворачивает хедер. Создание страниц - bem-tools создает пустой "desktop.bundles/contact/contact.bemjson.js"

И загадка что именно не работает, нужно перелопатить кучу документации чтобы понять все зависимости сборщика.

Было бы не плохо в корне project-stub иметь краткое описание настроек для сборки проекта:
 - ENB   //Сборщик, собирает проект из . Настраивается в файле .bem/enb-make.js
 - i-bem.js // Добавляет классы узлам Node в DOM-дереве, отключить/включить - в таком то месте.
 - и т.д.

Не претендую на реализацию. Но было бы проще, чем искать это по гитхабу в куче репозиториев.

P.S. видео не работает

**Comment by tadatuta** [comment]
Проблемы с туториалами в том, что на bem.info они чуть отстали от кода, в самом репозитории обновленный вариант. Вскоре выкатим на сайт.

> В ваших презентациях - красивый граф был

Он откроется именно в браузере, если открыть `http://localhost:8080/`.

> видео не работает

Сейчас проверил, у меня работает.

**Comment by DjonyBastone** [comment]
Да, вы правы - красивый граф открывается, и да по нему боле-менее понятно что под капотом. Спасибо

А видео не открывается - ошибка, говорится что-то о идентификаторе воспроизведения. С разных устройств попробовал, со сбросом куки и истории. Напишите название видео - сам попробую найти, или ссылку замаскированную.

**Comment by tadatuta** [comment]
Название: Автоматизация БЭМ
Заливший пользователь: artwtv 
Описание: Мастер-класс Владимира Гриненко на Frontend Talks&Works

**Comment by Realetive** [comment]
@DjonyBastone: https://youtu.be/-un-YYgU6Pg

P. S. Я даже был на этом мастер-классе ^__^

**Comment by DjonyBastone** [comment]
Спасибо всем
